diff --git a/ui/voip_calls.c b/ui/voip_calls.c
index 4ab23e18ef..2bad03a56b 100644
--- a/ui/voip_calls.c
+++ b/ui/voip_calls.c
@@ -65,8 +65,10 @@
 #include "ui/ws_ui_util.h"
 #include "ui/voip_calls.h"
 
-#define DUMP_PTR1(p) printf("#=> %p\n",(void *)p)
-#define DUMP_PTR2(p) printf("==> %p\n",(void *)p)
+#define DUMP_PTR1(p) fprintf(stderr, "#=> %p\n",(void *)p)
+#define DUMP_PTR2(p) fprintf(stderr, "==> %p\n",(void *)p)
+
+#define simple_dialog(...)
 
 const char *voip_call_state_name[8]={
     "",
diff --git a/sharkd.c b/sharkd.c
index 66b0694ec7..5af58ff57b 100644
--- a/sharkd.c
+++ b/sharkd.c
@@ -246,6 +246,17 @@ sharkd_get_frame_ts(void *data, guint32 frame_num)
   return NULL;
 }
 
+static const char *
+sharkd_get_user_comment(void *data, const frame_data *fd)
+{
+  capture_file *cf = (capture_file *) data;
+
+  if (cf->frames_user_comments)
+     return (const char *) g_tree_lookup(cf->frames_user_comments, fd);
+
+  return NULL;
+}
+
 static epan_t *
 sharkd_epan_new(capture_file *cf)
 {
@@ -255,7 +266,7 @@ sharkd_epan_new(capture_file *cf)
   epan->get_frame_ts = sharkd_get_frame_ts;
   epan->get_interface_name = cap_file_get_interface_name;
   epan->get_interface_description = cap_file_get_interface_description;
-  epan->get_user_comment = NULL;
+  epan->get_user_comment = sharkd_get_user_comment;
 
   return epan;
 }
@@ -769,6 +780,35 @@ sharkd_filter(const char *dftext, guint8 **result)
   return framenum;
 }
 
+static int
+frame_cmp(gconstpointer a, gconstpointer b, gpointer user_data _U_)
+{
+  const frame_data *fdata1 = (const frame_data *) a;
+  const frame_data *fdata2 = (const frame_data *) b;
+
+  return (fdata1->num < fdata2->num) ? -1 :
+    (fdata1->num > fdata2->num) ? 1 :
+    0;
+}
+
+const char *
+sharkd_get_comment(const frame_data *fd)
+{
+  return sharkd_get_user_comment(&cfile, fd);
+}
+
+int
+sharkd_set_comment(frame_data *fd, const gchar *new_comment)
+{
+  if (!cfile.frames_user_comments)
+    cfile.frames_user_comments = g_tree_new_full(frame_cmp, NULL, NULL, g_free);
+
+  fd->flags.has_user_comment = 1;
+  g_tree_replace(cfile.frames_user_comments, fd, g_strdup(new_comment));
+
+  return 0;
+}
+
 #include "version.h"
 const char *sharkd_version(void)
 {
diff --git a/sharkd.h b/sharkd.h
index 7b75788290..ad19b96441 100644
--- a/sharkd.h
+++ b/sharkd.h
@@ -22,6 +22,8 @@ int sharkd_retap(void);
 int sharkd_filter(const char *dftext, guint8 **result);
 int sharkd_dissect_columns(int framenum, column_info *cinfo, gboolean dissect_color);
 int sharkd_dissect_request(unsigned int framenum, void (*cb)(epan_dissect_t *, proto_tree *, struct epan_column_info *, const GSList *, void *), int dissect_bytes, int dissect_columns, int dissect_tree, void *data);
+const char *sharkd_get_comment(const frame_data *fd);
+int sharkd_set_comment(frame_data *fd, const gchar *new_comment);
 const char *sharkd_version(void);
 
 /* sharkd_daemon.c */
