diff --git a/epan/proto.c b/epan/proto.c
index f57e4873f5..671671c35e 100644
--- a/epan/proto.c
+++ b/epan/proto.c
@@ -6970,7 +6970,7 @@ static void
 tmp_fld_check_assert(header_field_info *hfinfo)
 {
 	gchar* tmp_str;
-
+return;
 	/* The field must have a name (with length > 0) */
 	if (!hfinfo->name || !hfinfo->name[0]) {
 		if (hfinfo->abbrev)
@@ -7506,7 +7506,7 @@ proto_register_field_init(header_field_info *hfinfo, const int parent)
 
 		/* Check that the filter name (abbreviation) is legal;
 		 * it must contain only alphanumerics, '-', "_", and ".". */
-		c = wrs_check_charset(fld_abbrev_chars, hfinfo->abbrev);
+		c = 0;
 		if (c) {
 			if (g_ascii_isprint(c))
 				fprintf(stderr, "Invalid character '%c' in filter name '%s'\n", c, hfinfo->abbrev);
diff --git a/epan/uat_load.l b/epan/uat_load.l
index 8c8af5452f..60891cee42 100644
--- a/epan/uat_load.l
+++ b/epan/uat_load.l
@@ -372,6 +372,7 @@ uat_load(uat_t *uat, char **errx)
 		return TRUE;
 	}
 
+	return FALSE; /* no such file */
 
 	if (!(in = ws_fopen(fname,"r"))) {
 		*errx = g_strdup(g_strerror(errno));
diff --git a/epan/geoip_db.c b/epan/geoip_db.c
index d4de2ccb99..4570f423d1 100644
--- a/epan/geoip_db.c
+++ b/epan/geoip_db.c
@@ -120,7 +120,7 @@ geoip_dat_scan_dir(const char *dirname) {
             name = ws_dir_get_name(file);
             if (g_str_has_prefix(file, "Geo") && g_str_has_suffix(file, ".dat")) {
                 datname = g_strdup_printf("%s" G_DIR_SEPARATOR_S "%s", dirname, name);
-                gi = GeoIP_open(datname, GEOIP_MEMORY_CACHE);
+                gi = GeoIP_open(datname, GEOIP_STANDARD);
                 if (gi) {
                     g_array_append_val(geoip_dat_arr, gi);
                 }
diff --git a/sharkd.c b/sharkd.c
index 8ca9ff328f..f3e435737e 100644
--- a/sharkd.c
+++ b/sharkd.c
@@ -792,6 +792,9 @@ const char *sharkd_version(void)
 #endif
 }
 
+/* newer glibc have memcpy@@GLIBC_2.14, and openshift don't have it */
+void *memcpy(void *dst, const void *src, size_t len) { return memmove(dst, src, len); }
+
 /*
  * Editor modelines  -  https://www.wireshark.org/tools/modelines.html
  *
diff --git a/ui/tap-sequence-analysis.c b/ui/tap-sequence-analysis.c
index eda3b049f8..27c4442b8b 100644
--- a/ui/tap-sequence-analysis.c
+++ b/ui/tap-sequence-analysis.c
@@ -344,7 +344,7 @@ sequence_analysis_list_free(seq_analysis_info_t *sainfo)
 
     /* free the graph data items */
 
-#if GLIB_CHECK_VERSION (2, 32, 0)
+#if 0
        g_queue_free_full(sainfo->items, sequence_analysis_item_free);
        sainfo->items = g_queue_new();
 #else
